dodiscover/ci/categorical_test.py

```diff
@@ -0,0 +1,200 @@
+import logging
+from functools import reduce
+from typing import Optional, Set, Tuple
+
+import pandas as pd
+from scipy import stats
+
+from dodiscover.ci.base import BaseConditionalIndependenceTest
+from dodiscover.typing import Column
+
+
+# copied from pgmpy
+def power_divergence(X, Y, Z, data, lambda_="cressie-read"):
```

Comment:
Types are missing.

Also, if you copied this from pgmpy, pay attention to their license (is it ok to copy and modify in the first place) and accordingly reference the license if required.

Reply:
pgmpy is MIT licensed so it should be fine, I believe

Reply:
You still need to copy the license over accordingly:
https://github.com/pgmpy/pgmpy/blob/dev/LICENSE#L12

So, just copying the code without that would still not be ok.

Reply:
But all pywhy packages are MIT licensed, so I think this should be okay to just leave as is and provide credit assignment to pgmpy in the docstrings?

Reply:
Strictly speaking, there is "Copyright (c) 2013-2021 pgmpy" in their MIT license, which we also strictly speaking need to copy 1:1.
Ok, at least emphasize that part and say it follows MIT license.